add block-share-card extension metafile

qcrao opened the pull request:
This PR introduces the Block Share Card extension for Roam Research. 

This extension allows users to easily share content blocks from Roam Research as images optimized for mobile and desktop viewing. The functionality enhances the user experience by providing a quick and efficient way to share information in a visually appealing format.

baibhavbista commented:
@qcrao looks like a really cool extension!

I had some concerns/feature requests:
(I used your extension to get my notes to a form that you can easily view! So the extension is coming in handy already!)
![Baibhav Bista-DGlDWFmlH-desktop (1)](https://github.com/Roam-Research/roam-depot/assets/23657255/e52423a6-7932-4114-8345-84899249e779)

qcrao commented:
@baibhavbista 

Thank you so much for taking the time to review my extension and provide feedback! I'm thrilled to hear that you've already found it useful in converting your notes into a more accessible format.

I noticed you mentioned having some concerns and feature requests. Could you please elaborate on what those are? I'd love to learn more about your thoughts so I can work on addressing any issues and enhancing the extension based on your suggestions.

Your input is invaluable in helping me improve the user experience. I'm all ears and looking forward to hearing more details!

baibhavbista commented:
@qcrao You're welcome 😁

btw if you have not used it/run into it before, you can use `extensionAPI` to create settings panel
documentation here: https://roamresearch.com/#/app/developer-documentation/page/4xiXPZuEm

qcrao commented:
@baibhavbista 

Thank you for the updates! Here's the revised summary of your improvements:

1. Fixed the issue where the name is too long and looks bad in the mobile view.
2. Added a settings panel with the following options:
- Card style: Allows selecting between "Default" and "Vanilla Roam" styles.
- Show/hide blocks and days information: Provides an option to toggle the display of the blocks and days count in the footer.
- Display create time or edit time in the header.
3. Removed the unnecessary "content" div from the header and also removed the corresponding CSS code.
4. Updated the file links in the readme file, ensuring that they are now displayed correctly.

Please review the changes again. Thank you for your time and consideration.

---

Below are some example images showcasing the changes:

<img width="1188" alt="image" src="https://github.com/Roam-Research/roam-depot/assets/7698088/71b6ae8b-23ff-444e-8216-c6adf96fbeeb">

![desktop](https://github.com/Roam-Research/roam-depot/assets/7698088/fa4a3d9a-0ff9-4824-8f3a-0f095a7bbbcf)

![mobile](https://github.com/Roam-Research/roam-depot/assets/7698088/168b0503-cb68-4bff-880c-61e62026b2e5)

baibhavbista commented:
Looks great @qcrao , will release it now!
